### PR TITLE
EVG-7183 ignore empty diffs with force updates

### PR DESCRIPTION
**Pull request by ablack12**

### CODE REVIEW

File: service/api_patch.go [hunk 1]
```diff
@@ -166,7 +167,8 @@ func (as *APIServer) updatePatchModule(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	if p.Alias == evergreen.CommitQueueAlias && !patch.IsMailboxDiff(data.PatchString) {
+	data.PatchString = strings.TrimSpace(data.PatchString)
+	if p.Alias == evergreen.CommitQueueAlias && len(data.PatchString) != 0 && !patch.IsMailboxDiff(data.PatchString) {
```
Comment (ybrill) [inline service/api_patch.go]: I don't think there's a use case for an empty module patch

File: service/api_patch.go [hunk 2]
```diff
@@ -53,12 +53,13 @@ func (as *APIServer) submitPatch(w http.ResponseWriter, r *http.Request) {
 	if patchString == "" {
 		patchString = data.PatchString
 	}
+	patchString = strings.TrimSpace(patchString)
```
Comment (ybrill) [inline service/api_patch.go]: are we concerned for patches that contain only space?

Comment (ablack12) [reply service/api_patch.go]: I'm not particularly concerned about that: don't diffs usually contain lines before and after also? I was mostly trying to circumvent something weird happening where the patch string wasn't empty but should be. If you think that check is overkill I don't mind dropping it.

File: service/api_patch.go (hunk 1 above)
Comment (ablack12) [reply service/api_patch.go]: Fair but I'd rather remain consistent

File: service/api_patch.go (hunk 2 above)
Comment (ablack12) [reply service/api_patch.go]: Failing tests have convinced me to drop it :)

**Comment by ablack12** [comment]
Verbal LGTM from Jonathan, and validated in staging.

**Pull request merged**